### ISSUE TITLE
ffmpeg raspberry pi jessie install or use avconv

### ISSUE THREAD

**Issue by totomtornado** [PHP-FFMpeg/PHP-FFMpeg#299]
| Q              | A
| -------------- | ---
| Bug?           | no
| New Feature?   | no
| Version Used   | Specific tag or commit sha
| FFmpeg Version | FFmpeg or AVConv and version
| OS             | Your OS and version (raspbian Jessie)

#### Actual Behavior

How does PHP-FFMpeg behave at the moment?
I have tried everything to get ffmpeg and php-ffmpeg to work together.  I am kind of a novice in the linux world but am trying.  I got this to work for a short time.  I shut off the computer and came back and it did not work again.  My stopping point is no matter what install I try of ffmpeg, I get this error....

`Fatal error: Uncaught exception 'Alchemy\BinaryDriver\Exception\ExecutableNotFoundException' with message 'Executable not found, proposed : /usr/share/ffprobe' in /home/pi/src/vendor/alchemy/binary-driver/src/Alchemy/BinaryDriver/AbstractBinary.php:160 Stack trace: #0 /home/pi/src/vendor/php-ffmpeg/php-ffmpeg/src/FFMpeg/Driver/FFProbeDriver.php(48): Alchemy\BinaryDriver\AbstractBinary::load('/usr/share/ffpr...', NULL, Object(Alchemy\BinaryDriver\Configuration)) #1 /home/pi/src/vendor/php-ffmpeg/php-ffmpeg/src/FFMpeg/FFProbe.php(207): FFMpeg\Driver\FFProbeDriver::create(Array, NULL) #2 /home/pi/src/vendor/php-ffmpeg/php-ffmpeg/src/FFMpeg/FFMpeg.php(117): FFMpeg\FFProbe::create(Array, NULL, NULL) #3 /var/www/html/video_builder/working-thumbs.php(8): FFMpeg\FFMpeg::create(Array) #4 {main} Next exception 'FFMpeg\Exception\ExecutableNotFoundException' with message 'Unable to load FFProbe' in /home/pi/src/vendor/php-ffmpeg/php-ffmpeg/src/FFMpeg/Driver/FFProbeDriver.php:50 Stack trace: #0 /home/pi/src/vendor/php-ffmpeg/php-ffmp in /home/pi/src/vendor/php-ffmpeg/php-ffmpeg/src/FFMpeg/Driver/FFProbeDriver.php on line 50`

I tried setting the binary paths in my php like:

```php
$ffmpeg = FFMpeg\FFMpeg::create(array(
    'ffmpeg.binaries'  => '/usr/share/ffmpeg',
    'ffprobe.binaries' => '/usr/share/ffprobe',
    'timeout'          => 3600, // The timeout for the underlying process
    'ffmpeg.threads'   => 12,   // The number of threads that FFMpeg should use
));
```

#### Expected Behavior

I just want to use these two together to get some thumbs from some videos on the server.

Honestly, I am kind of lost at this point and the code above is perhaps useless as I have installed and reinstalled and removed ffmpeg so many times now... also I have tried reinstalling php-ffmpeg through composer and uninstalled it many times.

I am just curious if there is any advice on how to continue... Now I am wondering if this library works with avconv.  I realize ffmpeg is not available for jessie without backports or some outside build(?)... a little over my head but I gather I can not simply use sudo apt-get and obtain it....

any advice on how to move forward would be great!   Perhaps I need to clean up my installs first?  Currently it seems that all ffmpeg installs are out because:

> ffmpeg -version returns >> -bash: /usr/bin/ffmpeg: No such file or directory

I uninstalled php-ffmpeg...

I am just worried because I did try installing with instructions here and although it worked and took a century... I could never get php-ffmpeg working
https://www.assetbank.co.uk/support/documentation/install/ffmpeg-debian-squeeze/ffmpeg-debian-jessie/

Could I have an extra install hiding somewhere... should I use avconv instead?

**Comment by totomtornado** [comment]
Long post I know....

It seems so tough to get solid info on ffmpeg install for jessie... after thought I am considering a second machine(another pi3B) to use with this wrapper.  This fits what I want to do with my project a little better anyway.

I am contemplating using ubuntu MATE on this new pi to run a second LAMP and ultimately ffmpeg and this wrapper.  Any experiences with ubuntu MATE, ffmpeg, and this wrapper used together?  Can anyone give feedback on positive or negative experiences using this setup?

Is there a better lightweight option I am missing, perhaps?

Cheers...

**Comment by Romain** [comment]
Hi @totomtornado,

It seems clear that the executable is not found, so the problem comes from the way you installed FFMPEG on your machine.

I don't know the guide you follow, but I can only recommend that you follow the [official guide](https://trac.ffmpeg.org/wiki/CompilationGuide/), which is cristal clear and works like a charm.

Let me know if this helps.

**Comment by totomtornado** [comment]
right... it says "Executable not found, proposed : /usr/share/ffprobe"... which is because I specified the path to the ffprobe and ffmpeg in the script I wrote BUT... the thing is that the files were in that location so....

All in all... My searching showed me way too many ways to compile or install ffmpeg on raspbian jessie and so many more errors or problems with it on this system.  That is why I was curious about ubuntu mate... 

Like I said in my 2nd post above... another pi makes sense in my setup anyway.  I plan on using the 2nd pi just for making video thumbnails and as my testing area.  With the new pi, I plan on using a different OS this time around... Ubuntu MATE has ffmpeg on its repository and is fine for what I want to do with it so I am hoping for a smoother approach.  Any feedback on how that works with someone else would be awesome.  

To someone more comfortable with linux systems this may all sound stupid but I am a bit of a novice.  Regardless it seems safer to go with an OS that officially supports ffmpeg and has it on their repository rather than workaround through the back in something like raspbian Jessie...

If none of it works then I guess I will just use a plain old exec call and run it from the command line out of php... I would love to use the wrapper though as I assume it is better in many ways else why make a wrapper...???

So... Still wondering... Anyone?  Anyone?
Have you used this with Ubuntu Mate on a pi with minimal hassle?  I will try to post back in a week or so with my experience regardless.

**Comment by Romain** [comment]
I've no experience with Ubuntu MATE.
However, if your executable cannot be found, and exec won't work either...

**Comment by totomtornado** [comment]
I was able to get my new pi and try out a Ubuntu Mate install with this wrapper.

I am happy to report it works very well.

I copied over the image and used Ubuntu's Tasksel to install my LAMP stack
https://help.ubuntu.com/community/Tasksel

Then I got ffmpeg apt-get... the package is available on the repository and fine for me but for other situations you still may need to compile, etc.
https://ubuntu-mate.community/t/tutorial-build-or-download-ffmpeg-libavcodec-with-mmal-support-hardware-acceleration-video-decoding/3565

I installed composer and then got the wrapper as suggested here.

No fuss... working fine.... 

I recommend this over using Jessie... It was much simpler and straightforward if you are not an expert with linux or the terminal stuff yet.
